src/problem.jl

```diff
@@ -0,0 +1,82 @@
+"""
+    StateSpaceProblem{isinplace, dType, uType,tType,noiseType,oType, lType} <: AbstractStateSpaceProblem{isinplace}
+
+A state space problem contains all the information necessary to simulate a state space
+process.
+"""
+struct StateSpaceProblem{isinplace, dType, uType,tType,noiseType,oType, lType} <: AbstractStateSpaceProblem{isinplace}
+    f::Function # Evolution
+    g::Function # Noise function
+    h::Function # Observation function
```

Comment:
Make this concretely typed and add type parameters?

Reply:
Ah, good catch. I'm never quite sure whether these should be the same type parameters or different ones. They have different types (i.e. `typeof(f) != typeof(g)` since function types are weird).

Probably the easiest practice is different type parameters, though I always hate structs with a bajillion type params.

Reply:
Would the Linear specific specialization be a different type with the same `AbstractStateSpaceProblem`?

Reply:
Yeah, the current `StateSpaceProblem` might be better renamed as `NonlinearStateSpaceProblem` or something, since it allows for arbitrary transition functions. I'll add a `LinearStateSpaceProblem` that's much more specialized with the accompanying `_solve` methods.

Reply:
I would just do StateSpaceProblem and show that it's linear by putting a linear operator in for `f` (`AbstractDiffEqOperator`, that interface should be sufficient?)